### DISCUSSION TITLE
Do not consider empty metric a fail

### DISCUSSION THREAD

**Discussion by aminbs** [testspace-com/support#15] [Ideas]
Hello,
We are currently using testspace for our monorepo and different checks with metrics generation are ran for different apps for CI optimisation.

This means that depending on the commit a metric result may not be pushed, it would be very helpful for us to be able to not consider empty results as failures.

Thanks

**Comment by ivailop** [comment]
@aminbs, thanks for reporting that issue. It appears we have a bug with "Code Coverage" metrics. 
Do you see the problem with any other metric?

**Reply by aminbs** [reply]
Hi @ivailop this is about analysis metric, we enforce it as pass/fail but we would like the option to consider no result a pass
As we have many checks and run them based on conditions in our repo.
Any help would be appreciated 

<img width="132" alt="Screenshot 2021-04-24 at 00 14 52" src="https://user-images.githubusercontent.com/2975648/115934768-19cd5680-a492-11eb-816b-696fa22f3063.png">
<img width="470" alt="Screenshot 2021-04-24 at 00 15 17" src="https://user-images.githubusercontent.com/2975648/115934786-28b40900-a492-11eb-9163-7baef6ba600f.png">
<img width="556" alt="Screenshot 2021-04-24 at 00 15 48" src="https://user-images.githubusercontent.com/2975648/115934828-3b2e4280-a492-11eb-8831-49a9513534bd.png">
<img width="846" alt="Screenshot 2021-04-24 at 00 16 08" src="https://user-images.githubusercontent.com/2975648/115934843-471a0480-a492-11eb-9978-1b40a70233c8.png">

**Comment by ivailop** [comment]
That appears to be related to the requested with https://github.com/testspace-com/testspace.support/issues/1 enhancement. When we addressed it, apparently, we didn't think of the missing metrics use-case... 

We'll try to address it similarly with the upcoming release - i.e. CI `success` would suppress invalid (aka "orange") check for missing metric's dataset.

**Reply by aminbs** [reply]
indeed, thank you

**Comment by ivailop** [comment]
@aminis, the issue should be fixed now.

**Reply by aminbs** [reply]
works great, thanks to the team !
